extensions/machine-learning/src/views/models/prediction/columnsTable.ts

```diff
@@ -382,6 +383,44 @@ export class ColumnsTable extends ModelViewBase implements IDataComponent<Predic
 		return warningButton;
 	}
 
+	public openWarningCalloutDialog(dialogHeading: string, dialogName?: string, calloutMessageText?: string, calloutMessageLinkText?: string, calloutMessageLinkUrl?: string): void {
+		// Included offset values for Mac client.
+		let macClientXYOffset: azdata.window.IDialogXYOffset = { xOffset: 304, yOffset: 30 };
+		let dialog = azdata.window.createModelViewDialog(dialogHeading, dialogName, 'narrow', 'calloutCompact', 'left', false, true, macClientXYOffset);
+		let warningTab: azdata.window.DialogTab = azdata.window.createTab('tab1');
+		warningTab.registerContent(async view => {
+			let warningContentContainer = view.modelBuilder.divContainer().withProperties({
+				CSSStyles: {}
+			}).component();
+			let messageTextComponent = view.modelBuilder.text().withProperties({
+				value: calloutMessageText,
+				CSSStyles: {
+					'font-size': '12px',
+					'line-height': '16px',
+					'margin': '0 0 12px 0'
+				}
+			}).component();
+			warningContentContainer.addItem(messageTextComponent);
+
+			if (calloutMessageLinkText && calloutMessageLinkUrl) {
+				const messageLinkComponent = view.modelBuilder.hyperlink().withProperties({
+					label: calloutMessageLinkText,
+					url: calloutMessageLinkUrl,
+					CSSStyles: {
+						'font-size': '13px',
+						'margin': '0px'
+					}
+				}).component();
+				warningContentContainer.addItem(messageLinkComponent);
+			}
+			view.initializeModel(warningContentContainer);
+		});
+		// set tab as content
+		dialog.content = [warningTab];
+
+		azdata.window.openDialog(dialog);
```

Comment:
@Charles-Gagnon After this is called, the warning dialog appears. However, the top portion appears first, followed by the remaining content -- it's slow enough to observe. Attached is a video of the behavior. I do not know where to get at the functionality in order to make changes. Placing a breakpoint here resulted in me stepping through a collection 249 promises. There has got to be a more efficient way of observing WHY the dialog renders the second half later. Can you offer any insight here?

https://user-images.githubusercontent.com/12279161/102650929-786b7800-4120-11eb-8856-cc3934c3df1e.mov

Reply:
@Charles-Gagnon @alanrenmsft I sure could use your help with improving this behavior.

Reply:
Is this something you need to do in this PR? Given the huge size of this PR and how long it's been out it'd be nice to get this out and then come back and file issues with a repro that can be done and reviewed separately.

Reply:
@Charles-Gagnon No, I can create an issue for this.